### DISCUSSION TITLE
load other image types beyond `tga` and `png`

### DISCUSSION THREAD

**Discussion by xarthurx** [libigl/libigl#1766] [Q&A] [answered]
This might be a dummy question, but I cannot find an answer anywhere in the issue lists.

There're functions `texture_from_file` that load `png` image into OpenGL texture:
https://github.com/libigl/libigl/blob/main/include/igl/png/texture_from_file.h

But why it only supports `png` (and `tga`) type but not other common ones?

Specifically, why no `jpeg`? Is it because `jpeg` is somewhat compressed and cannot be read directly?

**Comment by jdumas** [comment, marked as answer]
libigl uses [stb_image](https://github.com/nothings/stb) to read an image file. So technically all image format supported by stb_image should be supported, included jpg.

**Comment by xarthurx** [comment]
OK, I check the code and it actually does.

However, isn't the name of the function misleading?
`texture_from_png` can read image formats that not only is `png`;
`texture_from_file` calls the above function, but only limits the format to `png`...

This is very confusing...

if `libigl` is meant to support all kinds of format, should the `igl::png` model also be modified/re-structured to make it more clear?

**Reply by jdumas** [reply]
Yes.
